### PR TITLE
[WIP] [feature] Create Helm chart for driver

### LINKED ISSUE #28: [feature] Create Helm chart for driver
here is the reference: https://github.com/ritazh/secrets-store-csi-driver/tree/master/charts/keyvault-csi-driver

### PR DESCRIPTION
**Pull request by lizebang**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Create Helm chart for driver. I have tested it.

**Which issue(s) this PR fixes**:

Fixes #28 

**Special notes for your reviewer**:

1. I have not added the label `app: {{ .Release.Name }}` to the template "adcd.labels". Because some places used the `app` label. I don't know if the label `app` is managed separately or centrally.

2. How to manage Chart SemVer `version`? Is it simply the same as `appversion`?

**Release note**:
```
[feature] Create Helm chart for driver
```

### CODE REVIEW

File: charts/azuredisk-csi-driver/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+appVersion: "0.1.0-alpha"
```
Comment (andyzhangx) [inline charts/azuredisk-csi-driver/Chart.yaml]: let's use `v0.2.0-alpha`

File: charts/azuredisk-csi-driver/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+description: Azure disk Container Storage Interface (CSI) Storage Plugin
+name: azuredisk-csi-driver
+version: 0.1.0-alpha
```
Comment (andyzhangx) [inline charts/azuredisk-csi-driver/Chart.yaml]: this is chart version, use `1.0.0`?

File: charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+          livenessProbe:
+            failureThreshold: 5
+            httpGet:
+              path: /healthz
+              port: healthz
+            initialDelaySeconds: 30
+            timeoutSeconds: 10
+            periodSeconds: 30
+          env:
+{{- if eq .Values.provider "azure" }}
```
Comment (andyzhangx) [inline charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml]: I think this driver is only for azure, so remove it?

File: charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  namespace: {{ .Release.Namespace }}
+{{ include "adcd.labels" . | indent 2 }}
+    app: csi-azuredisk
+spec:
+  selector:
+    matchLabels:
+      app: csi-azuredisk
+  template:
+    metadata:
+{{ include "adcd.labels" . | indent 6 }}
```
Comment (andyzhangx) [inline charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml]: what dose `adcd` mean?

**Comment by andyzhangx** [comment]
> I have not added the label `app: {{ .Release.Name }}` to the template "adcd.labels". Because some places used the `app` label. I don't know if the label `app` is managed separately or centrally.

Not sure about the answer, but you may refer to https://github.com/helm/charts/blob/master/stable/wordpress/Chart.yaml#L2-L3

 > How to manage Chart SemVer `version`? Is it simply the same as `appversion`?

It's seperate, you could refer to https://github.com/helm/charts/blob/master/stable/wordpress/Chart.yaml#L2-L3

And also, could your write a README for how to build, and use chart for this driver, thanks.

File: charts/azuredisk-csi-driver/Chart.yaml (hunk 1 above)
Comment (lizebang) [reply charts/azuredisk-csi-driver/Chart.yaml]: OK.
keep `appVersion` = `image version`?

File: charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml (hunk 4 above)
Comment (lizebang) [reply charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml]: Emmm...
`adcd = azure disk csi driver`, maybe it's not intuitive.

File: charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml (hunk 3 above)
Comment (lizebang) [reply charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml]: I will do it.

File: charts/azuredisk-csi-driver/Chart.yaml (hunk 2 above)
Comment (lizebang) [reply charts/azuredisk-csi-driver/Chart.yaml]: I don't think `1.0.0` is ok. This does not comply with the semver specification. [Please see](https://semver.org/#how-do-i-know-when-to-release-100). `0.1.0` or `0.2.0` might be better.

**Comment by lizebang** [comment]
> And also, could your write a README for how to build, and use chart for this driver, thanks.

Yes, I will do it. And can I write it into [docs/install-azuredisk-csi-driver.md](https://github.com/csi-driver/azuredisk-csi-driver/blob/master/docs/install-azuredisk-csi-driver.md)？Thanks.

File: charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml (hunk 4 above)
Comment (andyzhangx) [reply charts/azuredisk-csi-driver/templates/azuredisk-csi-driver.yaml]: maybe `azuredisk` is better

File: charts/azuredisk-csi-driver/Chart.yaml (hunk 2 above)
Comment (andyzhangx) [reply charts/azuredisk-csi-driver/Chart.yaml]: ok, then use `0.1.0`

File: charts/azuredisk-csi-driver/Chart.yaml (hunk 1 above)
Comment (andyzhangx) [reply charts/azuredisk-csi-driver/Chart.yaml]: ` appVersion` here app means application, right? then it should be image version.

**Comment by andyzhangx** [comment]
> Yes, I will do it. And can I write it into [docs/install-azuredisk-csi-driver.md](https://github.com/csi-driver/azuredisk-csi-driver/blob/master/docs/install-azuredisk-csi-driver.md)？Thanks.

I think you could write a `README.md` in `charts` folder, and then add a link in  `docs/install-azuredisk-csi-driver.md`

File: charts/azuredisk-csi-driver/Chart.yaml (hunk 1 above)
Comment (lizebang) [reply charts/azuredisk-csi-driver/Chart.yaml]: Yes

**Comment by lizebang** [comment]
I will do it.

**Comment by andyzhangx** [comment]
pls also create helm chart for azure file driver, thanks.
https://github.com/csi-driver/azurefile-csi-driver/issues/32

**Pull request merged**